### PR TITLE
Video tabs

### PR DESCRIPTION
**Pull request by yichenshen**
This PR adds tabs for videos.

The motivation was that the video list for CS1010X was getting too long and had to be categorized.

For the most part, it's a simple grouping for videos, with the UI similar to that of assessment tabs.

### CODE REVIEW

File: app/controllers/components/course/videos_component.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def main_sidebar_items
         icon: 'video-camera',
         title: settings.title || t('course.video.videos.sidebar_title'),
         weight: 12,
-        path: course_videos_path(current_course),
+        path: course_videos_path(current_course, tab: current_course.video_tabs.first),
```
Comment (weiqingtoh) [inline app/controllers/components/course/videos_component.rb]: `course.video_tabs.first` is used quite often. Can we abstract it into a separate method?

**Comment by yichenshen** [comment]
Yup, I think all of your suggestions here makes sense, will try to fix them up asap.

**Comment by yichenshen** [comment]
@weiqingtoh Updated as suggested

File: app/controllers/course/video/videos_controller.rb [hunk 2]
```diff
@@ -38,6 +46,12 @@ def destroy
...
 
   def video_params
-    params.require(:video).permit(:title, :description, :start_at, :url, :published)
+    params.require(:video).permit(:title, :tab_id, :description, :start_at, :url, :published)
+  end
+
+  def current_tab
+    return @video.tab if @video&.tab.present?
+    return @tab = Course::Video::Tab.find(params[:tab]) if params[:tab].present?
+    @tab = current_course.default_video_tab
```
[the remark this answers is not in the document]
Comment (yichenshen) [reply app/controllers/course/video/videos_controller.rb]: Ah ok, my bad, I actually meant to do that.

**Comment by yichenshen** [comment]
Alright, should be updated.

**Pull request merged**